symphony/lib/toolkit/class.timestampvalidator.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * @package toolkit
+ */
+
+/**
+ * This class checks against the Database for the latest timestamp, to make
+ * sure the data being saved is the one the user saw.
+ */
+
+class TimestampValidator
+{
+    /**
+     * The name of the table to check against
+     * @var string
+     */
+    private $table = '';
+
+    /**
+     * Creates a new TimestampValidator object, for a particular table.
+     * The `tbl_` prefix is automatically added.
+     * @param [string] $table the table name
+     */
+    public function __construct($table)
+    {
+        assert($table, 'Table cannot be falsy');
```

Comment:
The behaviour of assert depends on [php configuration](http://php.net/assert) values. The idea is usually that development code will throw errors, but production instances optimise away the calls.

I'd prefer regular throwing in this scenario.